test/e2e/serverless_operator_test.go

```diff
@@ -0,0 +1,86 @@
+package e2e
+
+import (
+	"testing"
+
+	"github.com/openshift-knative/serverless-operator/test"
+	"github.com/pkg/errors"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+const (
+	serviceName        = "knative-openshift-metrics"
+	serviceMonitorName = serviceName
+)
+
+func TestServerlessOperator(t *testing.T) {
+	caCtx := test.SetupClusterAdmin(t)
+
+	test.CleanupOnInterrupt(t, func() { test.CleanupAll(t, caCtx) })
+
+	t.Run("create subscription and wait for CSV to succeed", func(t *testing.T) {
+		if _, err := test.WithOperatorReady(caCtx, "serverless-operator-subscription"); err != nil {
+			t.Fatal("Failed", err)
+		}
+	})
+
+	// Check the status of the serverless operator deployment
+	if err := test.CheckDeploymentScale(caCtx, test.OperatorsNamespace, "knative-openshift", 1); err != nil {
+		t.Fatalf("Failed to verify the operator deployment: %v", err)
+	}
+
+	// Check if service monitors are installed
+	if _, err := caCtx.Clients.Kube.CoreV1().Services(test.OperatorsNamespace).Get(serviceName, metav1.GetOptions{}); err != nil {
+		t.Fatalf("Failed to get the operator monitoring service : %v", err)
+	}
+
+	if _, err := caCtx.Clients.MonitoringClient.ServiceMonitors(test.OperatorsNamespace).Get(serviceMonitorName, metav1.GetOptions{}); err != nil {
+		t.Fatalf("Failed to verify the operator service monitor: %v", err)
+	}
+
+	t.Run("undeploy serverless operator and check dependent monitoring resources removed", func(t *testing.T) {
+		caCtx.Cleanup(t)
+		if err := waitForOperatorMonitoringServiceDeleted(caCtx); err != nil {
+			t.Fatalf("Monitoring service is still available: %v", err)
+		}
+		if err := waitForOperatorServiceMonitorDeleted(caCtx); err != nil {
+			t.Fatalf("Service monitor is still available: %v", err)
+		}
+	})
+}
```

Comment:
This is because in the past we migrated olm stuff and there were some env vars missing meant eg. check [this](https://github.com/openshift-knative/serverless-operator/pull/442). We discussed it and we agreed we need an e2e test.
I think I need the subscription to setup the operator no?

Reply:
There are two packages, e2e and serving/eventinge2e. The former assumes that no operator is installed yet and it has to be setup completely.
If you want to be part of that codebase, I'd propose to hook into the existing `TestKnativeEventing` and `TestKnativeServing` tests (which could arguably be collapsed together at this point too to save some time, but that's separate), as that'll shave quite a bit of runtime.

Reply:
Ok I will move tests there.